tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino

```diff
@@ -353,6 +353,59 @@ void PmsInit(void) {
   }
 }
 
+// This gives more accurate data for forest fire smoke.  PurpleAir gives you this conversion option labeled "US EPA"
+// https://cfpub.epa.gov/si/si_public_record_report.cfm?dirEntryId=353088&Lab=CEMM
+/*
+Copy-paste from the PDF Slide 26
+y={0 ≤ x <30: 0.524*x - 0.0862*RH + 5.75}
+y={30≤ x <50: (0.786*(x/20 - 3/2) + 0.524*(1 - (x/20 - 3/2)))*x -0.0862*RH + 5.75}
+y={50 ≤ x <210: 0.786*x - 0.0862*RH + 5.75}
+y={210 ≤ x <260: (0.69*(x/50 – 21/5) + 0.786*(1 - (x/50 – 21/5)))*x - 0.0862*RH*(1 - (x/50 – 21/5)) + 2.966*(x/50 – 21/5) + 5.75*(1 - (x/50 – 21/5)) + 8.84*(10^{-4})*x^{2}*(x/50 – 21/5)}
+y={260 ≤ x: 2.966 + 0.69*x + 8.84*10^{-4}*x^2}
+
+y= corrected PM2.5 µg/m3
+x= PM2.5 cf_atm (lower)
+RH= Relative humidity as measured by the PurpleAir
+*/
+int usaEpaStandardPm2d5Adjustment(int pm25_standard, int relative_humidity)
+{
+  // Rename to use the same variables from the paper
+  float x = pm25_standard;
+  float RH = relative_humidity;
+  if (x<30) {
+    return 0.524 * x - 0.0862 * RH + 5.75;
```

Comment:
All literals must be forced to floats using `f` suffix, like `0.524f`. Otherwise computation is done using double, which eats up a lot of flash.